### DISCUSSION TITLE
Is it possible to use AutoOffsetReset: earliest and provide an offset or timestamp to reset to?

### DISCUSSION THREAD

**Discussion by RossSteele** [LGouellec/streamiz#356] [Q&A] [answered]
Just the question in the title, is it possible to reset to a specific offset or timestamp, or does earliest go to start of the retention period?

Thanks

**Comment by LGouellec** [comment, marked as answer]
Hey @RossSteele , 

Not by design directly in the application. The `AutoOffsetReset` policy is only used if your consumer group does not have a valid offset committed. 

If you would like to reset the offset for a given consumer group and partition, you can use the Kafka CLI `kafka-consumer-groups.sh` to reset the offsets : 
Example : 
``` bash
kafka-consumer-groups.sh --bootstrap-server localhost:9092 --group myConsumerGroup --reset-offsets --to-earliest --topic my_topic -execute
```

You can seek per offset or per timestamp or beginning or latest.
